src/components/directory-content.jsx

```diff
@@ -412,434 +341,142 @@ const DirectoryContent = () => {
-                                size={18}
-                                sx={styles.circularRoot}
-                            />
-                        )}
-                    {childrenMetadata[element.elementUuid] &&
-                        getFileIcon(element.type, styles.icon)}
-                    {/* Name */}
-                    <OverflowableText
-                        text={getDisplayedElementName(cellData)}
-                        tooltipSx={styles.tooltip}
-                    />
-                </Box>
-            );
-        },
-        [childrenMetadata, currentChildren, getDisplayedElementName]
-    );
-
-    function toggleSelection(elementUuid) {
-        let element = currentChildren?.find(
-            (e) => e.elementUuid === elementUuid
-        );
-        if (element === undefined) {
-            return;
-        }
-        let newSelection = new Set(selectedUuids);
-        if (!newSelection.delete(elementUuid)) {
-            newSelection.add(elementUuid);
-        }
-        setSelectedUuids(newSelection);
-    }
-
-    function toggleSelectAll() {
-        if (selectedUuids.size === 0) {
-            setSelectedUuids(
-                new Set(
-                    currentChildren
-                        .filter((e) => !e.uploading)
-                        .map((c) => c.elementUuid)
-                )
-            );
-        } else {
-            setSelectedUuids(new Set());
-        }
-    }
-
-    function selectionHeaderRenderer() {
-        return (
-            <Box
-                onClick={(e) => {
-                    toggleSelectAll();
-                    e.stopPropagation();
-                }}
-                sx={styles.checkboxes}
-            >
-                <Checkbox
-                    checked={selectedUuids.size > 0}
-                    indeterminate={
-                        selectedUuids.size !== 0 &&
-                        selectedUuids.size !== currentChildren.length
-                    }
-                />
-            </Box>
-        );
-    }
-
-    function selectionRenderer(cellData) {
-        const elementUuid = cellData.rowData['elementUuid'];
-        return (
-            <Box
-                onClick={(e) => {
-                    toggleSelection(elementUuid);
-                    e.stopPropagation();
-                }}
-                sx={styles.checkboxes}
-            >
-                <Checkbox checked={selectedUuids.has(elementUuid)} />
-            </Box>
-        );
-    }
-
     useEffect(() => {
         if (!selectedDirectory) {
             return;
         }
         setIsMissingDataAfterDirChange(true);
     }, [selectedDirectory, setIsMissingDataAfterDirChange]);
 
-    useEffect(() => {
-        if (!currentChildren?.length) {
-            setChildrenMetadata({});
-            setIsMissingDataAfterDirChange(false);
-            return;
-        }
+    const [checkedElement, setCheckedElement] = useState([]);
```

Comment:
create states at the beginning of components when possible

Reply:
Done, renamed it checkedRows by the way